kerastuner/engine/oracle.py

```diff
@@ -17,90 +17,241 @@
 from __future__ import division
 from __future__ import print_function
 
+import collections
 import enum
 import hashlib
+import json
 
+from tensorflow import keras
+from kerastuner.engine import hyperparameters as hp_module
+from kerastuner.engine import metrics_tracking
+from kerastuner.engine import stateful
+from kerastuner.engine import trial as trial_lib
+from ..abstractions.tensorflow import TENSORFLOW_UTILS as tf_utils
 
-class Oracle(object):
 
-    def __init__(self):
-        self.space = []
+Objective = collections.namedtuple('Objective', 'name direction')
 
-    def update_space(self, new_entries):
+
+class Oracle(stateful.Stateful):
+
+    def __init__(self,
+                 objective,
+                 max_trials,
```

Comment:
Please make this an optional argument. If unset, tuning should continue until the user stops it.
Users may also wish to limit tuning by wall-clock time, but that can be added later.

Reply:
Done, not clear how to handle `remaining_trials` then, but for now it will return `None` if not applicable